tests/main/snap-on-non-shared-root/task.yaml

```diff
@@ -22,12 +22,12 @@ execute: |
     test-snapd-tools.echo hello
 
     echo "Ensure we have a shared mount of $SNAPMOUNTDIR"
-    cat /proc/self/mountinfo |MATCH "$SNAPMOUNTDIR $SNAPMOUNTDIR.*shared:1"
+    cat /proc/self/mountinfo |MATCH "$SNAPMOUNTDIR $SNAPMOUNTDIR.*shared:[0-9]*"
```

Comment:
Sure, this is the log, 
https://paste.ubuntu.com/24634598/
https://paste.ubuntu.com/24634600/

I just can reproduce it when I use -reuse, otherwise I can't reproduce it. It is like it works the first time when allways the group 1 is assigned to /snap, but in the following executions you get other group ids and there is no match.